.azure-devops/templates/build-publish-azure-cli-test-sdk.yml

```diff
@@ -0,0 +1,21 @@
+steps:
+  - script: 'rm -rf azure-cli'
+    displayName: 'delete azure cli directory'
+
+  - script: 'git clone https://github.com/Azure/azure-cli.git'
```

Comment:
with my limited knowledge, does pipeline always to clean clone, in the sense of it would not try to pull and then clone if needed?
It seems not needed at this point, but maybe you want a clone resursive just in case for the future.

Reply:
I'm leaning towards lets add --recursive when we need it rather than preemptively.